backend/app/requests/caching.py

```diff
@@ -0,0 +1,81 @@
+"""Methods to cache adequacy and representative points responses."""
+import json
+import logging
+import os
+import six
+
+import ujson
+
+from backend.config import config
+
+logger = logging.getLogger(__name__)
+
+CACHE_DIRECTORY = config.get('cached_result_directory')
+
+
+def cache(hint_fields):
+    """
+    Decorator function for caching API responses.
+
+    If caching is disabled, responses are not cached.
+    If any of the hint fields are empty or missing, responses are not cached.
+
+    Otherwise, attempt to return previously cached responses, saving the responses
+    whose files do not yet exist.
+    """
+    def wrap(f):
+        @six.wraps(f)
+        def wrapped_f(**kwargs):
+            return _cache(func=f, hint_fields=hint_fields, **kwargs)
+        return wrapped_f
+    return wrap
+
+
+def _cache(func, hint_fields, **kwargs):
+    hint_values = [kwargs.get(field) for field in hint_fields]
+    cache_filepath = _get_cached_filepath(
+        prefix=func.__name__,
+        hint_values=hint_values,
+    )
+    # If caching is disabled or a hint is missing, call the function normally.
+    if not all(hint_values) or not config.get('cache_requests'):
+        response = func(**kwargs)
+    # If the file exists, read and return.
+    elif os.path.isfile(cache_filepath):
+        with open(cache_filepath, 'r') as f:
+            response = ujson.load(f)
```

Comment:
Maybe invert this and return directly.
Then you can do response = func(**kwargs)
And then do test to cache or not, no?

Reply:
I initially had inverted this, but it gets a bit tricky with the boolean expressions in the `if...elif`. We could do the following:

1. Check if caching is enabled and filepath exists. 
2. If so, return cached response.
3. If not, calculate `response = func(**kwargs)`. 
4. Check if caching is enabled. If so, write to file. If not, return.

Not sure that's any cleaner though.

Reply:
But actually we could have `use_cache` and `add_to_cache` as separate variables. We could disable `add_to_cache` on the public version but still have `use_cache` set to `True`.